client-core/src/service/key_service.rs

```diff
@@ -1,38 +1,70 @@
 use secstr::SecUtf8;
-use zeroize::Zeroize;
 
-use client_common::{PrivateKey, PublicKey, Result, SecureStorage, Storage};
-
-const KEYSPACE: &str = "core_key";
+use super::basic_key_service::BasicKeyService;
+use super::hdkey_service::HDKeyService;
+use super::key_service_data::{KeyServiceInterface, WalletKinds};
+use client_common::{PrivateKey, PublicKey, Result, Storage};
 
 /// Maintains mapping `public-key -> private-key`
 #[derive(Debug, Default, Clone)]
 pub struct KeyService<T: Storage> {
-    storage: T,
+    kind: WalletKinds,
+    basic: Option<BasicKeyService<T>>,
+    hd: Option<HDKeyService<T>>,
```

Comment:
or change Key service to `enum`? Does KeyService where `basic` and `hd` are both None make sense?

Reply:
i'm working on using one KeyService
basic, hd will be removed.